motoman_gp200r_support/urdf/gp200r_macro.xacro

```diff
@@ -0,0 +1,164 @@
+            </geometry>
+        </collision>
+    </link>
+    <link name="${prefix}link_6_t">
+        <visual>
+            <geometry>
+                <mesh filename="package://motoman_gp200r_support/meshes/gp200r/visual/link_6_t.STL" scale="0.001 0.001 0.001"/>
+            </geometry>
+            <material name="yaskawa_blue"/>
+        </visual>
+        <collision>
+            <geometry>
+                <mesh filename="package://motoman_gp200r_support/meshes/gp200r/visual/link_6_t.STL" scale="0.001 0.001 0.001"/>
+            </geometry>
+        </collision>
+    </link>
+    <link name="${prefix}tool0"/>
+    <!-- end of link list -->
+
+    <!-- joint list -->
+    <joint name="${prefix}joint_1_s" type="revolute">
+        <parent link="${prefix}base_link"/>
+        <child link="${prefix}link_1_s"/>
+        <origin xyz="0 0 0.450" rpy="0 0 0" />
+        <axis xyz="0 0 1" />
+        <limit lower="${radians(-180)}" upper="${radians(180)}" effort="9414.38" velocity="${radians(125)}"/>
+    </joint>
+    <joint name="${prefix}joint_2_l" type="revolute">
+        <parent link="${prefix}link_1_s"/>
+        <child link="${prefix}link_2_l"/>
+        <origin xyz="0.740 0 0" rpy="0 ${radians(90)} 0" />
+        <axis xyz="0 1 0" />
+        <limit lower="${radians(-130)}" upper="${radians(80)}" effort="11645.4" velocity="${radians(115)}"/>
+    </joint>
+    <joint name="${prefix}joint_3_u" type="revolute">
+        <parent link="${prefix}link_2_l"/>
+        <child link="${prefix}link_3_u"/>
+        <origin xyz="0 0 1.150" rpy="0 0 0" />
+        <axis xyz="0 -1 0" />
+        <limit lower="${radians(-78.4)}" upper="${radians(78)}" effort="7060.79" velocity="${radians(125)}"/>
+    </joint>
+    <joint name="${prefix}joint_4_r" type="revolute">
+        <parent link="${prefix}link_3_u"/>
+        <child link="${prefix}link_4_r"/>
+        <origin xyz="0 0 0.250" rpy="0 0 0" />
+        <axis xyz="-1 0 0" />
+        <limit lower="${-3.14}" upper="${3.14}" effort="1392.54" velocity="${radians(182)}"/>
+    </joint>
+    <joint name="${prefix}joint_5_b" type="revolute">
+        <parent link="${prefix}link_4_r"/>
+        <child link="${prefix}link_5_b"/>
+        <origin xyz="1.225 0 0" rpy="0 0 0" />
+        <axis xyz="0 -1 0" />
+        <limit lower="${radians(-125)}" upper="${radians(125)}" effort="1863.26" velocity="${radians(175)}"/>
+    </joint>
+    <joint name="${prefix}joint_6_t" type="revolute">
+        <parent link="${prefix}link_5_b"/>
+        <child link="${prefix}link_6_t"/>
+        <origin xyz="0.250 0 0" rpy="0 0 0" />
+        <axis xyz="1 0 0" />
+        <limit lower="${-3.14}" upper="${3.14}" effort="926.73" velocity="${radians(265)}"/>
+    </joint>
+    <joint name="${prefix}joint_6_t-tool0" type="fixed">
+        <origin xyz="0 0 0" rpy="${radians(180)} ${radians(-90)} 0"/>
+        <parent link="${prefix}link_6_t"/>
+        <child link="${prefix}tool0"/>
+    </joint>
+    <!-- end of joint list -->
+
```

Comment:
We might also want to add a flange frame here, to work towards https://github.com/ros-industrial/motoman/issues/374. Refer to [motoman_hc10_support/urdf/hc10dt_b10_macro.xacro](https://github.com/ros-industrial/motoman/blob/kinetic-devel/motoman_hc10_support/urdf/hc10dt_b10_macro.xacro) for an example.

Reply:
I have added a flange frame as done in the referenced hc10 xacro.